### ISSUE TITLE
Problem due to wrong dimensions in E4TEncoder

### ISSUE THREAD

**Issue by Actis92** [mkshing/e4t-diffusion#6]
Hi, 
Thank you for your job. Trying your code I'm having problems during the pertaining due to wrong dimensions in E4TEncoder.
Because when you do
`clip_hidden_states = torch.cat(clip_hidden_states)
  clip_hidden_states = torch.mean(clip_hidden_states, dim=0)
  clip_hidden_states = clip_hidden_states.expand(1, -1)`
If you are working with a batch for example of 16, after these 3 lines, your first dimension is 1, because you are taking the average across the batch (using dim=0)
would not be better to do the mean in the dimension 1?
`clip_hidden_states = torch.cat(clip_hidden_states,dim=1)
        clip_hidden_states = torch.mean(clip_hidden_states, dim=1)
        clip_hidden_states = clip_hidden_states.unsqueeze(1)`
So that we can concatenate with the other features before the final linear layer? 
Obviously, with this change also the dimension of the final layer 
```
self.final_linear = nn.Linear(
            1+sum(block_out_channels),
            word_embedding_dim,
        )
```
Let me know what do you think!

**Comment by mkshing** [comment]
@Actis92 Hi, thank you for your feedback! I found this bug and fixed in https://github.com/mkshing/e4t-diffusion/commit/a0b96f0f2a7688934763f6d970ebb97454488832. Please let me know if you have further questions. Thanks.

**Comment by Actis92** [comment]
Thank you for the fix
